packages/common/src/sandbox/modules.ts

```diff
@@ -251,14 +252,30 @@ export const isMainModule = (
 export const findMainModule = (
   modules: Module[],
   directories: Directory[],
-  entry: string = 'index.js'
+  entry: string = 'index.js',
+  template?: TemplateType
 ) => {
   try {
+    // first attempt: try loading the entry file if it exists
     const module = resolveModule(entry, modules, directories);
-
     return module;
   } catch (e) {
```

Comment:
ugh nested try catch. Is there a better way to handle this? This looks ugly.

Reply:
Hmm, if we use `resolveModuleWrapped` here we can do a nullcheck on the result, move on to the next flow if needed.

Reply:
Nice, I really like that